src/common/graphql/graphql.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { ApiConfigService } from "../api-config/api.config.service";
+import { GraphQLClient } from 'graphql-request';
+
+@Injectable()
+export class GraphQlService {
+  private readonly logger: Logger;
+  private readonly MICROSERVICE_URL: string;
+  private readonly graphqlClient: GraphQLClient;
+
+  constructor(
+    private readonly apiConfigService: ApiConfigService
+  ) {
+    this.logger = new Logger(GraphQlService.name);
+    this.MICROSERVICE_URL = this.apiConfigService.getMicroServiceUrlMandatory();
```

Comment:
since this is only used in the constructor, you can use a local variable for microservice url